File: 프로그래머스/unrated/181868. 공백으로 구분하기 2/공백으로 구분하기 2.java
//1.일반 풀이 정규표현식. \\s는 공백에 관한 정규식+는 공백 한 개 이상인 경우라는 의미 이를 공백 하나로 바꾸라는 의미.
class Solution {
    public String[] solution(String my_string) {
        return my_string.trim().replaceAll("\\s+"," ").split(" ");
    }
}

File: 프로그래머스/unrated/181868. 공백으로 구분하기 2/공백으로 구분하기 2.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>
// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
char** solution(const char* my_string) {
    int word_cnt = 1;
    for(int i = 0; my_string[i] != NULL; i++){
        if(my_string[i] == '\\s'&&my_string[i+1] != '\\s'){
            word_cnt++;
        }        
    }
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    char** answer = (char**)malloc(sizeof(char*) * word_cnt);
    char* token = strtok(my_string, " ");
    int i = 0;
    while(token != NULL){
        //answer[i] = (char*)malloc(sizeof(char) * (strlen(token) + 1));
        answer[i++] = token;
        token = strtok(NULL, " ");
    }
         
    return answer;
}